src/core/lib/surface/call.cc

```diff
@@ -446,6 +430,22 @@ grpc_error* grpc_call_create(const grpc_call_create_args* args,
                                       &call->call_combiner};
   add_init_error(&error, grpc_call_stack_init(channel_stack, 1, destroy_call,
                                               call, &call_args));
+  // Publish this call to parent only after the call stack has been initialized.
+  if (args->parent != nullptr) {
+    child_call* cc = call->child;
+    parent_call* pc = get_or_create_parent_call(args->parent);
+    gpr_mu_lock(&pc->child_list_mu);
+    if (pc->first_child == nullptr) {
+      pc->first_child = call;
+      cc->sibling_next = cc->sibling_prev = call;
+    } else {
+      cc->sibling_next = pc->first_child;
+      cc->sibling_prev = pc->first_child->child->sibling_prev;
+      cc->sibling_next->child->sibling_prev =
+          cc->sibling_prev->child->sibling_next = call;
+    }
+    gpr_mu_unlock(&pc->child_list_mu);
+  }
```

Comment:
How certain are we that any of the grpc_call_create actions following this line  won't cause the same problem (= are we sure that the call is fully initialized when publishing it?).   The cancel_with_error  invocations seem harmless, but I'd doublecheck the `grpc_polling_entity_create_from_pollset` and `grpc_call_stack_set_pollset_or_pollset_set`

Reply:
Yeah, this PR only fixes the publish-before-init problem for refcount. But the early-use problem we've seen is also just about refcount, so I guess it's safer to just move the publishing code after the refcount init code.

(After moving the publishing code to just before returning, the refcount flake still exists.)